### ISSUE TITLE
Install Instructions aren't working for me

### ISSUE THREAD

**Issue by patrickjburke245** [skarnet/skalibs#7]
Steps to reproduce:

Spun up EC2 on AWS (OS = Amazon Linux, AMI = amzn2-ami-hvm-2.0.20230628.0-x86_64-gp2, size = t2.micro)
SSH to machine
Install git (sudo yum install git)
Install C compiler (sudo yum install gcc)
Clone git repo (git clone git://git.skarnet.org/skalibs)
Change directory into /skalibs (cd skalibs)
Run install instructions (./configure && make && sudo make install)
Getting error message "Makefile:12: *** Missing config.mak; please use ./configure first. Stop.".

Steps to troubleshoot: ran ./configure, then make (not as one conjoined command with "&&"). Still ran into same issue.

Please let me know what additional info I can provide.

Full output message:
Checking for C compiler... ... gcc Checking whether C compiler works...  ... yes Checking target system type... ... x86_64-redhat-linux Checking whether compiler accepts -fPIC ... ... yes Checking whether compiler accepts -std=c99 ... ... yes Checking whether compiler accepts -fomit-frame-pointer ... ... yes Checking whether compiler accepts -fno-exceptions ... ... yes Checking whether compiler accepts -fno-unwind-tables ... ... yes Checking whether compiler accepts -fno-asynchronous-unwind-tables ... ... yes Checking whether compiler accepts -Werror=implicit-function-declaration ... ... yes Checking whether compiler accepts -Werror=implicit-int ... ... yes Checking whether compiler accepts -Werror=pointer-sign ... ... yes Checking whether compiler accepts -Werror=pointer-arith ... ... yes Checking whether compiler accepts -Wno-unused-value ... ... yes Checking whether compiler accepts -Wno-parentheses ... ... yes Checking whether compiler accepts -ffunction-sections ... ... yes Checking whether compiler accepts -fdata-sections ... ... yes Checking whether linker accepts -nostartfiles ... ... yes Checking whether linker accepts -Wl,--as-needed ... ... yes Checking whether linker accepts -Wl,--sort-section=alignment ... ... yes Checking whether linker accepts -Wl,--sort-common ... ... yes Checking whether linker accepts -Wl,--hash-style=both ... ... yes Checking required linker flags for socket functions... ... none Checking whether system has clock_gettime()... ... yes Checking whether system has CLOCK_MONOTONIC... ... yes Checking whether system has CLOCK_BOOTTIME... ... yes Checking whether system has posix_spawn()... ... yes Checking whether system has timer_create()... ... yes, with linker args: -lrt Checking endianness... ... little Checking size and signedness of standard types... ... done Checking whether system has accept4()... ... yes Checking whether system has MSG_CMSG_CLOEXEC... ... yes Checking whether system has dirfd()... ... yes Checking whether system has eventfd()... ... yes Checking whether system has flock()... ... yes Checking whether system has getpeereid()... ... no Checking whether system has SO_PEERCRED... ... yes Checking whether system has getpeerucred()... ... no Checking whether system has IPv6 support... ... yes Checking whether system has MSG_DONTWAIT... ... yes Checking whether system has O_DIRECTORY... ... yes Checking whether system has openat()... ... yes Checking whether system has linkat()... ... yes Checking whether system has memmem()... ... yes Checking whether system has pipe2()... ... yes Checking whether system has ppoll()... ... yes Checking whether system has revoke()... ... no Checking whether system has sendfile()... ... yes Checking whether system has setgroups()... ... yes Checking whether system has settimeofday()... ... yes Checking whether system has signalfd()... ... yes Checking whether system has splice()... ... yes Checking whether system has strcasestr()... ... yes Checking whether system has strnlen()... ... yes Checking whether system has uint64_t... ... yes Checking whether system has futimens()... ... yes Checking whether system has futimes()... ... yes Checking whether system has arc4random()... ... no Checking whether system has arc4random_addrandom()... ... no Checking whether system has setitimer()... ... yes Checking whether system has namespaces... ... yes Checking whether system has NS_GET_PARENT... ... yes Checking whether system has explicit_bzero()... ... yes Checking whether system has getrandom()... ... yes Checking whether system has GRND_INSECURE... ... no Checking whether system has chroot()... ... yes Checking whether system has /dev/urandom... ... yes Checking whether system has posix_spawn() returning early... ... no Creating config.mak... ... done. Creating src/include/skalibs/config.h... ... done. Makefile:12: *** Missing config.mak; please use ./configure first.  Stop.

**Comment by skarnet** [comment]
Absolutely cannot reproduce. Sounds like a bug in make, or in the underlying filesystem. Please check your make version? or try `sync` between `configure` and `make`?

**Comment by patrickjburke245** [comment]
`sync ./configure ./make` did not fix the issue. Maybe I'm using that command incorrectly.

After running `make --version`, I receive this output:
```
GNU Make 3.82
Built for x86_64-koji-linux-gnu
Copyright (C) 2010  Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law
```.

Are there extra setup steps to make sure my system implements POSIX.1-2008?

**Comment by skarnet** [comment]
`./configure && sync && make`

But okay, that make is old and I haven't tested it in a while. Can you try with a more recent version of GNU make (4.0 to 4.4) and see if it fixes the issue? If that's the case, it means that I'm using a construct that isn't supported by old versions of make (which I'm still claiming to support) so I'll need to fix that.

**Comment by patrickjburke245** [comment]
Make has been updated to 4.4, and I was then able to run all 3 steps (./configure, make, sudo make install) without issue. Thank you very much!

For those who might encounter this issue in the future (by using a VM or container without a 4.0-4.4 version of Make), here are the steps to upgrade Make:
1. Pull a version of Make from the [GNU website](https://ftp.gnu.org/gnu/make/?C=M;O=D) onto your machine. Example: wget https://ftp.gnu.org/gnu/make/make-4.4.tar.gz
2. Extract the tarball. Example: tar -xf make-4.4.tar.gz
3. cd into make-4.4 directory
4. Follow its INSTALL instructions
5. Replace the default make version with this newly-built version. Example for Amazon Linux: mv ./make /usr/bin/make
6. Now return to the skalibs directory and retry its INSTALL steps.

I could not find an easy way for my package manager (yum) to install the 4.x family of make, thus why I had to follow these above steps.
